catalog-rest-service/src/main/java/org/openmetadata/catalog/jdbi3/PipelineServiceRepository.java

```diff
@@ -0,0 +1,139 @@
+import org.openmetadata.catalog.exception.EntityNotFoundException;
+import org.openmetadata.catalog.type.Schedule;
+import org.openmetadata.catalog.util.EntityUtil;
+import org.openmetadata.catalog.util.JsonUtils;
+import org.skife.jdbi.v2.sqlobject.Bind;
+import org.skife.jdbi.v2.sqlobject.CreateSqlObject;
+import org.skife.jdbi.v2.sqlobject.SqlQuery;
+import org.skife.jdbi.v2.sqlobject.SqlUpdate;
+import org.skife.jdbi.v2.sqlobject.Transaction;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.List;
+
+import static org.openmetadata.catalog.exception.CatalogExceptionMessage.entityNotFound;
+
+
+public abstract class PipelineServiceRepository {
+  private static final Logger LOG = LoggerFactory.getLogger(PipelineServiceRepository.class);
+
+  @CreateSqlObject
+  abstract PipelineServiceDAO pipelineServiceDAO();
+
+  @CreateSqlObject
+  abstract EntityRelationshipDAO relationshipDAO();
+
+  @Transaction
+  public List<PipelineService> list(String name) throws IOException {
+    return JsonUtils.readObjects(pipelineServiceDAO().list(name), PipelineService.class);
+  }
+
+  @Transaction
+  public PipelineService get(String id) throws IOException {
+    return EntityUtil.validate(id, pipelineServiceDAO().findById(id), PipelineService.class);
+  }
+
+  @Transaction
+  public PipelineService getByName(String name) throws IOException {
+    return EntityUtil.validate(name, pipelineServiceDAO().findByName(name), PipelineService.class);
+  }
+
+  @Transaction
+  public PipelineService create(PipelineService pipelineService) throws JsonProcessingException {
+    // Validate fields
+    validateIngestionSchedule(pipelineService.getIngestionSchedule());
+    pipelineServiceDAO().insert(JsonUtils.pojoToJson(pipelineService));
+    return pipelineService;
+  }
+
+  public PipelineService update(String id, String description, URI url,
+                                 Schedule ingestionSchedule)
+          throws IOException {
+    validateIngestionSchedule(ingestionSchedule);
+    PipelineService pipelineService = EntityUtil.validate(id, pipelineServiceDAO().findById(id), PipelineService.class);
+    // Update fields
+    pipelineService.withDescription(description).withIngestionSchedule(ingestionSchedule)
+            .withPipelineUrl(url);
+    pipelineServiceDAO().update(id, JsonUtils.pojoToJson(pipelineService));
+    return pipelineService;
+  }
+
+  @Transaction
+  public void delete(String id) {
+    if (pipelineServiceDAO().delete(id) <= 0) {
+      throw EntityNotFoundException.byMessage(entityNotFound(Entity.PIPELINE_SERVICE, id));
+    }
+    relationshipDAO().deleteAll(id);
+  }
+
+  private void validateIngestionSchedule(Schedule ingestion) {
+    if (ingestion == null) {
+      return;
+    }
+    String duration = ingestion.getRepeatFrequency();
+
+    // ISO8601 duration format is P{y}Y{m}M{d}DT{h}H{m}M{s}S.
```

Comment:
This can be a utility function used for other ingestion schedules as well?

Reply:
Agree. Added Utils.java and replaced it in all the repository classes